### PR TITLE
Updated cats and monix dependencies

### PR DESCRIPTION
**Pull request by ilya-murzinov**
As requested in #68 
Tests are failing right now, I'll fix them

### CODE REVIEW

File: shared/src/main/scala/implicits.scala [hunk 1]
```diff
@@ -17,16 +17,15 @@
...
 
 object implicits extends FutureInstances {
   implicit def fetchFutureFetchMonadError(
-      implicit ec: ExecutionContext,
-      ME: MonadError[Future, Throwable]
+      implicit ec: ExecutionContext
   ): FetchMonadError[Future] = new FetchMonadError[Future] {
     override def runQuery[A](j: Query[A]): Future[A] = j match {
-      case Sync(e) => ME.pureEval(e)
+      case Sync(e) => pure(e.value)
```
Comment (ilya-murzinov) [inline shared/src/main/scala/implicits.scala]: @raulraja thanks for the hint!
Is is possible that this change is causing tests to fail?
```
[info] - Every level of joined concurrent fetches is combined and batched *** FAILED ***
[info]   (1,1,6) did not equal (3,3,6) (FetchTests.scala:481)
[info] - The product of concurrent fetches of the same type implies everything fetched in a single batch *** FAILED ***
[info]   (1,1,4) did not equal (2,1,4) (FetchTests.scala:457)
[info] - Every level of sequenced concurrent of concurrent fetches is batched *** FAILED ***
[info]   (5,5,19) did not equal (3,3,19) (FetchTests.scala:508)
```

Comment (dialelo) [inline shared/src/main/scala/implicits.scala]: Thanks for taking the time to make these changes Ilya, I've made some changes to the fetch execution reporting that should make those errors go away in the `reporting` branch. I'm going to try to combine your work and mine and cut a release as soon as possible.

Comment (ilya-murzinov) [inline shared/src/main/scala/implicits.scala]: @dialelo ooooops, I didn't see that it' already implemented in your branch :(

**Comment by ilya-murzinov** [comment]
Seems that this PR is useless since it's already implemented in #66

**Comment by ilya-murzinov** [comment]
@dialelo I'll leave this PR open in case you'll find anything useful in it. But feel free to close it :)

**Pull request closed without merging**